lib/src/components/date_input/date_input.dart

```diff
@@ -1,319 +1,314 @@
+  final DateTime? minDate;
 
-  /// A callback, which provides the entered date, or `null`, if invalid.
-  ///
-  /// See the example in the [ZetaDateInput] documentation how to provide custom validations
-  /// in combination with `hasError` and `errorText`.
-  final void Function(DateTime?)? onChanged;
+  /// The maximum date allowed by the date input.
+  final DateTime? maxDate;
 
-  /// `datePattern` is needed for the date format validation as described here:
-  ///  https://pub.dev/documentation/intl/latest/intl/DateFormat-class.html
-  final String datePattern;
+  /// The validator passed to the text input.
+  /// Returns a string containing an error message.
+  ///
+  /// By default, the form field checks for if the date is within [minDate] and [maxDate] (if given).
+  /// It also checks for null values unless [requirementLevel] is set to [ZetaFormFieldRequirement.optional]
+  ///
+  /// If the default validation fails, [errorText] will be shown.
+  /// However, if [validator] catches any of these conditions, the return value of [validator] will be shown.
+  final String? Function(DateTime? value)? validator;
 
   @override
-  State<ZetaDateInput> createState() => _ZetaDateInputState();
+  State<ZetaDateInput> createState() => ZetaDateInputState();
   @override
   void debugFillProperties(DiagnosticPropertiesBuilder properties) {
     super.debugFillProperties(properties);
     properties
-      ..add(EnumProperty<ZetaWidgetSize>('size', size))
-      ..add(StringProperty('label', label))
-      ..add(StringProperty('hint', hint))
-      ..add(DiagnosticsProperty<bool>('enabled', enabled))
       ..add(DiagnosticsProperty<bool>('rounded', rounded))
-      ..add(DiagnosticsProperty<bool>('hasError', hasError))
+      ..add(ObjectFlagProperty<ValueChanged<DateTime?>?>.has('onChange', onChange))
+      ..add(DiagnosticsProperty<DateTime?>('initialValue', initialValue))
+      ..add(DiagnosticsProperty<bool>('disabled', disabled))
+      ..add(StringProperty('label', label))
+      ..add(StringProperty('hintText', hintText))
       ..add(StringProperty('errorText', errorText))
-      ..add(ObjectFlagProperty<void Function(DateTime? p1)?>.has('onChanged', onChanged))
-      ..add(StringProperty('datePattern', datePattern));
+      ..add(EnumProperty<ZetaWidgetSize>('size', size))
+      ..add(ObjectFlagProperty<String? Function(DateTime value)?>.has('validator', validator))
+      ..add(StringProperty('dateFormat', dateFormat))
+      ..add(DiagnosticsProperty<DateTime?>('minDate', minDate))
+      ..add(DiagnosticsProperty<DateTime?>('maxDate', maxDate));
   }
 }
 
-class _ZetaDateInputState extends State<ZetaDateInput> {
-  final _controller = TextEditingController();
-  late ZetaWidgetSize _size;
-  late final String _hintText;
+/// State for [ZetaDateInput]
+class ZetaDateInputState extends State<ZetaDateInput> implements ZetaFormFieldState {
+  // TODO(mikecoomber): add AM/PM selector inline.
```

Comment:
todo

Reply:
Was waiting on designs, should I just freestyle it?

Reply:
I see, sorry wasnt looking close enough. In situations like this I have been creating a Jira ticket and replacing 'mikecoomber' with the ticket number